src/app/reporting/Engine.cpp

```diff
@@ -489,10 +488,35 @@ void Engine::Run()
 
     if (allReadClean)
     {
-        InteractionModelEngine::GetInstance()->ReleaseClusterInfoList(mpGlobalDirtySet);
+        mGlobalDirtySet.ReleaseAll();
     }
 }
 
+bool Engine::MergeOverlappedAttributePath(ClusterInfo & aAttributePath)
+{
+    bool overlapped = false;
+    mGlobalDirtySet.ForEachActiveObject([&](auto * path) {
+        // If overlapped, we would skip this target path,
+        // --If targetPath is part of previous path, return true
+        // --If previous path is part of target path, update filedid and listindex and mflags with target path, return true
+        if (path->IsAttributePathSupersetOf(aAttributePath))
+        {
+            overlapped = true;
+            return Loop::Break;
+        }
+        if (aAttributePath.IsAttributePathSupersetOf(*path))
+        {
+            overlapped         = true;
+            path->mListIndex   = aAttributePath.mListIndex;
+            path->mAttributeId = aAttributePath.mAttributeId;
+            return Loop::Break;
+        }
+        return Loop::Continue;
+    });
+
+    return overlapped;
```

Comment:
Could also just return whether `ForEachActiveObject` returned Loop::Break and not have this extra variable we have to close over, right?

Reply:
yes, updated, thanks